### ISSUE TITLE
Infinite seek loop when a blocked segment is found at the end of the stream

### ISSUE THREAD

**Issue by defagos** [SRGSSR/srgmediaplayer-apple#103]
If a blocked segment extends until the end of the associated media, the player will perform an infinite seek loop when reaching it, leading to a crash.

### Issue type

Incorrect behavior and crash.

### Environment information

> The simulator is a valid device as well. If all versions or devices seem to be affected, simply enter 'Any'

* Library version: 6.1.2
* iOS / tvOS version: Any
* Device: Any

### Reproducibility

Always reproducible

### Steps to reproduce

1. Use one of the demos available.
1. Define a blocked segment which reaches the end of the media.
1. Play and seek into the blocked segment.

**Comment by defagos** [comment]
Here is what happens:

- The player reaches the blocked area and initiates a seek after it.
- The seek is made with a method which checks if there is a blocked segment at the seek position (to account for consecutive blocked segments). Since the player is at the end it finds the same segment again, which is blocked.
- And this continues over and over...

**Comment by defagos** [comment]
I updated the UTs and made two fix proposals:

1. The first one does not change how skip notifications are sent. It just ensures that recursive skips are fixed.
2. The second one is an improvement. Previously, if we had contiguous blocked segments, skips were made one segment at a time, leading to skip notifications being emitted for all of them. This second proposal improves the situation by emitting skip notifications only for the first blocked segment. The player then looks for the next playable seek location and resumes there, without emitting skip notifications for other blocked segments in between (which was not adding any value IMHO). I updated the UT accordingly.

We can discuss whether this second fix proposal makes sense or if the 1st one should be kept.
